C2GUILauncher/src/Mods/ModManager.cs

```diff
@@ -0,0 +1,158 @@
+﻿using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
+using Newtonsoft.Json.Serialization;
+using Octokit;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace C2GUILauncher.Mods
+{
+
+    class CoreMods
+    {
+        public const string GithubBaseURL = "https://github.com";
+
+        public const string PluginPath = "TBL\\Binaries\\Win64\\Plugins";
+        public const string ModCachePath = ".mod_cache";
+
+        public const string AssetLoaderPluginPath = $"{PluginPath}\\C2AssetLoaderPlugin.dll";
+        public const string ServerPluginPath = $"{PluginPath}\\C2ServerPlugin.dll";
+        public const string BrowserPluginPath = $"{PluginPath}\\C2BrowserPlugin.dll";
+
+        public const string AssetLoaderPluginURL = $"{GithubBaseURL}/Chiv2-Community/C2AssetLoaderPlugin/releases/latest/download/C2AssetLoaderPlugin.dll";
+        public const string ServerPluginURL = $"{GithubBaseURL}/Chiv2-Community/C2ServerPlugin/releases/latest/download/C2ServerPlugin.dll";
+        public const string BrowserPluginURL = $"{GithubBaseURL}/Chiv2-Community/C2BrowserPlugin/releases/latest/download/C2BrowserPlugin.dll";
+
+    }
+
+    class ModManager
+    {
+
+        public string RegistryOrg { get; }
+        public string RegistryRepoName { get; }
+        private GitHubClient Client { get; }
+        private HttpClient HttpClient { get; }
+        public List<Mod> Mods { get; }
+
+        public ModManager(string registryOrg, string registryRepoName)
+        {
+            RegistryOrg = registryOrg;
+            RegistryRepoName = registryRepoName;
+            Mods = new List<Mod>();
+
+            Client = new GitHubClient(new ProductHeaderValue("Chiv2-Unchained-Launcher"));
+            HttpClient = new HttpClient();
+        }
+
+        public async Task UpdateModsList()
```

Comment:
Not a huge fan of this method. I wrote it with for loops, I write it with lambdas, I liked neither. I like this more than the for loops though.

The WhenAll stuff is especially weird.  Makes me want to get a real FP library which would make all of this much more straight forward.